.github/workflows/benchmarks-run.yml

```diff
@@ -16,9 +16,35 @@ on:
         default: all
 
 jobs:
+  build:
+    if: github.repository == 'frequency-chain/frequency'
+    name: Build Benchmark Binary
+    runs-on: ubicloud-standard-16
+    container: ghcr.io/frequency-chain/frequency/ci-base-image:1.4.0
+    permissions:
+      contents: read
+    steps:
+      - name: Check Out Repo
+        uses: actions/checkout@v4
+        with:
+          ref: ${{github.event.inputs.branch}}
+
+      - name: Build Binary for Benchmarks
+        run: |
+          echo "Running benchmark binary builder..."
+          ./scripts/run_benchmarks.sh -b
+
+      - name: Upload Binary
+        uses: actions/upload-artifact@v4
+        with:
+          name: benchmark-binary
+          path: target/release/frequency
+          retention-days: 3
```

Comment:
Note, pending PR #2331 needs the runtime wasm blob in addition to the node binary; may as well upload it here